### PR TITLE
[doc] add ctc weight for attention rescoring & substitute batchnorm in convolution module with layernorm

### PR DESCRIPTION
**Pull request by whiteshirt0429**

### CODE REVIEW

File: wenet/transformer/asr_model.py [hunk 1]
```diff
@@ -417,6 +419,9 @@ def attention_rescoring(self,
             for j, w in enumerate(hyp):
                 score += decoder_out[i][j][w]
             score += decoder_out[i][len(hyp)][self.eos]
+            # add ctc score
+            score += log_add([cur_hyps[i][1][0], cur_hyps[i][1][1]
```
Comment (robin1001) [inline wenet/transformer/asr_model.py:423]: put log_add in _ctc_prefix_beam_search

File: wenet/transformer/convolution.py [hunk 2]
```diff
@@ -56,7 +56,7 @@ def __init__(self,
             groups=channels,
             bias=bias,
         )
-        self.norm = nn.BatchNorm1d(channels)
+        self.norm = nn.LayerNorm(channels)
```
Comment (robin1001) [inline wenet/transformer/convolution.py:59]: modify your commit message to cover this change or change it in a pull request.

File: wenet/transformer/asr_model.py [hunk 3]
```diff
@@ -332,7 +332,7 @@ def _ctc_prefix_beam_search(
                                reverse=True)
             cur_hyps = next_hyps[:beam_size]
         hyps = [x[0] for x in cur_hyps]
-        return hyps, encoder_out
+        return hyps, encoder_out, cur_hyps
```
Comment (robin1001) [inline wenet/transformer/asr_model.py:335]: hyps shares the same info with cur_hyps, it's better we return hyps with it's corresponding total score here. The return is like List[(hyp, score)], encoder_out.

**Review by robin1001 — CHANGES_REQUESTED** [review]
see the comment inline.

**Review by robin1001 — APPROVED** [review]
